### ISSUE TITLE
Creating collection hangs while the pulsar pod is killed

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#7429]
#### Steps/Code to reproduce:
1. Deploy milvus release named `milvus-chaos` with cluster mode
2. Start a thread to continuously create and drop collection
3. After a while, injected chaos that kills pulsar pod
4. Observe whether the last `create_collection` request is successful

#### Expected result:
The last `create_collection` request get timeout response

#### Actual results:
Hours passed, the request gets no response, and new `list_conllection` request get error
```
E0902 11:53:38.877663451 1280280 chttp2_transport.cc:1113]   Received a GOAWAY with error code ENHANCE_YOUR_CALM and debug data equal to "too_many_pings"
Addr [10.98.0.13:19530] list_collections
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1630554818.878043975","description":"Error received from peer ipv4:10.98.0.13:19530","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-09-02 11:52:43.863707', 'RPC start': '2021-09-02 11:52:43.865096', 'RPC error': '2021-09-02 11:53:38.878662'}
```

#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93):
  bd3056f  Wed Sep  1 10:31:55 UTC 2021
- Deployment mode(standalone or cluster):
   cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
   pymilvus              2.0.0rc6.dev3
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

#### Additional context:
Server logs:
[client.log](https://github.com/milvus-io/milvus/files/7096358/client.log)

[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/7096300/milvus_logs.tar.gz)
 From the logs of rootCoord, after pulsar restarted. the roorCoord has successfully connected to the pulsar.
```
5321 time="2021-09-02T03:11:32Z" level=info msg="[TCP connection established]" local_addr="10.97.5.244:59868" remote_addr="pulsar://milvus-chaos-pulsar:6650"
5322 time="2021-09-02T03:11:32Z" level=info msg="[Connection is ready]" local_addr="10.97.5.244:59868" remote_addr="pulsar://milvus-chaos-pulsar:6650"
5352 time="2021-09-02T03:11:35Z" level=info msg="[Reconnected producer to broker]" cnx="10.97.5.244:59868 -> 10.96.142.90:6650" producerID=53 producer_name=standalone-0-52 topic="persistent://public/default/rootcoord-dml_52"
```
But did not process the previous request for collection creation.
Part of proxy's log:
```
1507 [2021/09/02 03:10:23.718 +00:00] [DEBUG] [impl.go:239] ["HasCollection enqueue"] [role=Proxy-] [db=] [collection=CreateChecker__KCNFguXT]
1508 [2021/09/02 03:10:23.719 +00:00] [DEBUG] [id.go:145] ["IDAllocator pickCanDoFunc"] [need=1] [total=198888] [remainReqCnt=0]
1509 [2021/09/02 03:10:23.719 +00:00] [DEBUG] [impl.go:253] [HasCollection] [role=Proxy-] [msgID=427439475347162201] [timestamp=427439482123583489] [db=] [collection=CreateChecker__KCNFguXT]
1510 [2021/09/02 03:10:23.720 +00:00] [DEBUG] [impl.go:260] ["HasCollection Done"] [] [role=Proxy-] [msgID=427439475347162201] [timestamp=427439482123583489] [db=] [collection=CreateChecker__KCNFguXT]
1511 [2021/09/02 03:10:23.727 +00:00] [DEBUG] [impl.go:132] ["CreateCollection enqueue"] [role=Proxy-] [db=] [collection=CreateChecker__KCNFguXT] [schema="ChdDcmVhdGVDaGVja2VyX19LQ05GZ3VYVCILEgVpbnQ2NBgBKAUiCRIFZmxvYXQoCiIcEgxmbG9hdF92ZWN0b3IoZTIKCgNkaW0SAzEyOA=="]
1512 [2021/09/02 03:10:23.728 +00:00] [DEBUG] [id.go:145] ["IDAllocator pickCanDoFunc"] [need=1] [total=198887] [remainReqCnt=0]
1513 [2021/09/02 03:10:23.728 +00:00] [DEBUG] [impl.go:145] [CreateCollection] [role=Proxy-] [msgID=427439475347162202] [timestamp=427439482136690689] [db=] [collection=CreateChecker__KCNFguXT] [schema="ChdDcmVhdGVDaGVja2VyX19LQ05GZ3VYVCILEgVpbnQ2NBgBKAUiCRIFZmxvYXQoCiIcEgxm     bG9hdF92ZWN0b3IoZTIKCgNkaW0SAzEyOA=="]
1514 [2021/09/02 03:10:24.240 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
```
Part of rootCoord's log:
```
1604 [2021/09/02 03:10:23.720 +00:00] [DEBUG] [root_coord.go:1273] [HasCollection] [name=CreateChecker__KCNFguXT] [msgID=427439475347162201]
1605 [2021/09/02 03:10:23.720 +00:00] [DEBUG] [root_coord.go:1293] ["HasCollection Success"] [name=CreateChecker__KCNFguXT] [msgID=427439475347162201]
1606 [2021/09/02 03:10:23.728 +00:00] [DEBUG] [root_coord.go:1204] ["CreateCollection "] [name=CreateChecker__KCNFguXT] [msgID=427439475347162202]
1607 [2021/09/02 03:10:23.728 +00:00] [DEBUG] [task.go:131] ["collection name -> id"] ["collection name"=CreateChecker__KCNFguXT] [collection_id=427439482136690689] ["default partition id"=427439482136690690]
1608 time="2021-09-02T03:10:24Z" level=warning msg="[Failed to connect to broker.]" error="dial tcp 10.96.142.90:6650: connect: connection refused" remote_addr="pulsar://milvus-chaos-pulsar:6650"
```

**Comment by sunby** [comment]
`CreateCollection` does not have a timeout param. And RootCoord will send the request to a msgstream. After pulsar crash, it will hang to wait for pulsar restarting. But it seems that client did not send the request after pulsar restarted successfully.

**Comment by sunby** [comment]
This issue has been fixed by #7428 . @ThreadDao

**Comment by ThreadDao** [comment]
This issue is fixed, and I will close it first.
```
Version:   db8067d
Built:     Fri Sep  3 07:44:18 UTC 2021
```
